source/common/http/filter_manager.cc

```diff
@@ -344,13 +344,13 @@ bool ActiveStreamDecoderFilter::canContinue() {
   // continue to further filters. A concrete example of this is a filter buffering data, the
   // last data frame comes in and the filter continues, but the final buffering takes the stream
   // over the high watermark such that a 413 is returned.
-  return !parent_.stopDecoderFilterChain();
+  return !parent_.stopDecoderFilterChain() && !parent_.state_.recreated_stream_;
```

Comment:
I'm confused as to why this is needed.

Generally canContinue should return true if we want to continue the filter chain, and false if we've sent a local response or done recreateStream or something.  

if parent_.state_.recreated_stream_ is true, 
parent_.state_.decoder_filter_chain_aborted_ should also have been set two lines above in recreateStream.  

given bool stopDecoderFilterChain() { return state_.decoder_filter_chain_aborted_; }
that would return true, canContinue would be false, which is as it should be.  What is the desired behavior change here?

Reply:
Yes, in recreateStream, `parent_.state_.decoder_filter_chain_aborted_` has already been set to true, and there is no need to check `parent_.state_.recreated_stream_`. My original intention for adding it here was to make the behavior similar to that in `ActiveStreamEncoderFilter::canContinue()` , where continuation is only allowed under the clear condition that `parent_.state_.recreated_stream_` is false.

But now I understand your preference to leave recreate stream out of this logic, which makes the judgment process here simpler. I will make adjustments accordingly

Reply:
good catch. I didn't notice the relationship beween the` recreate_stream_` and `decoder_filter_chain_aborted_`. After a re-check to the source code, I think rather than the `recreate_stream_`, the `encoder_filter_chain_aborted_` should may be used here.